### PR TITLE
[Doc] [runtime env] Clarify conditions for local `pip` and `conda` requirements files

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
We support specifying `{"pip": "requirements.txt"}`, but it isn't clear in the docs where `"requirements.txt"` is supposed to be located for this API to work.  This PR clarifies that it must be a local file, and that if a relative filepath is used, it should be relative to where the Ray script is being run, not relative to the path specified by `working_dir`. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by akshay-anyscale — APPROVED** [review]

**Review by sihanwang41 — APPROVED** [review]

**Comment by architkulkarni** [comment]
Hi @angelinalg, do you mind reviewing this on behalf of ray-docs when you get a chance?

File: doc/source/ray-core/handling-dependencies.rst [hunk 1]
```diff
@@ -351,9 +351,10 @@ The ``runtime_env`` is a Python dictionary or a Python class :class:`ray.runtime
 
   - Example: ``{"packages":["tensorflow", "requests"], "pip_check": False, "pip_version": "==22.0.2;python_version=='3.8.11'"}``
 
-  When specifying a ``requirements.txt`` file, referencing local files `within` that file is not supported (e.g. ``-r ./my-laptop/more-requirements.txt``, ``./my-pkg.whl``).
+  When specifying a path to a ``requirements.txt`` file, the file must be present on your local machine and it must be a valid absolute path or relative filepath relative to your local current working directory, *not* relative to the `working_dir` specified in the `runtime_env`.
+  Furthermore, referencing local files `within` a `requirements.txt` file is not supported (e.g. ``-r ./my-laptop/more-requirements.txt``, ``./my-pkg.whl``).
```
Comment (angelinalg) [inline doc/source/ray-core/handling-dependencies.rst:355]:
```suggestion
  Furthermore, referencing local files `within` a `requirements.txt` file is not supported (e.g., ``-r ./my-laptop/more-requirements.txt``, ``./my-pkg.whl``).
```

**Review by angelinalg — CHANGES_REQUESTED** [review]
LGTM

**Comment by architkulkarni** [comment]
@angelinalg thanks, updated!

**Comment by architkulkarni** [comment]
@angelinalg would you mind approving this PR? It still has "changes requested"

**Review by angelinalg — APPROVED** [review]
LGTM

**Pull request merged**